ixmp/core.py

```diff
@@ -164,6 +164,48 @@ def scenario_list(self, default=True, model=None, scen=None):
         return pd.DataFrame(self._backend.get_scenarios(default, model, scen),
                             columns=FIELDS['get_scenarios'])
 
+    def export_timeseries_data(self, path, default=True, model=None,
+                               scenario=None, variables=None):
+        """Bulk timeseries export to CSV file.
+
+        Parameters
+        ----------
+        file : os.PathLike
+            File name to export data to; must have the suffix '.csv'.
+
+            Result file will contain the following columns:
+
+            - model
+            - scenario
+            - version
+            - variable
+            - unit
+            - region
+            - meta
+            - time
+            - year
+            - value
+        default : bool, optional
+            :obj:`True` to include only TimeSeries versions marked as default.
+        model : str or None, optional
```

Comment:
Is this correct? Can't it be a list of str? cf. https://github.com/iiasa/ixmp/blob/fa5139b7659d416abb3f5fe79c5f09139d02779b/ixmp/backend/jdbc.py#L385

Similar with the `scenario` arg. `variables` should be "list of str"

Reply:
> Is this correct? Can't it be a list of str? cf.

As far as I see it is correct. The method signature mimics `get_scenarios` which can also take model/scenario names as parameters expecting string or None to be passed.

Reply:
I see—so the functionality provided for `model`/`scenario` and for `variable` are not the same?
1. For `model`, you can optionally provide a single model name to restrict to only that model.
2. For `scenario`, the same.
3. For `variable`, you can provide a list of one or more names?

If that's the case, the docstring should be like:
```
model: str, optional
    Only return data for this model name.
scenario: str, optional
    Only return data for this scenario name.
variable: list of str, optional
    Only return data for variable name(s) in this list.
```
I also think `variable` is better than `variables` as an arg name.

Reply:
> I also think variable is better than variables as an arg name.

Didn't get the idea why it is better, but if you think so I'll change it

Reply:
For example, when we handle a `filters` dictionary, if the dimension is named 'technology' or 'year', then 'technology'/'year' is what one uses as a key in that dict…even if it maps to a list of technologies or years (plural). `variable` is, essentially, another dimension, so IMO should be treated the same way.